### DISCUSSION TITLE
Trouble connecting to coturn

### DISCUSSION THREAD

**Discussion by terraslate** [sipsorcery-org/sipsorcery#1157] [General]
I've installed coturn, chrome connects to it and sends data channel messages between two peers on the same computer. I used a very basic username and password.  I have put the same details for my sipsorcery connections but it doesn't seem to work. I debugged it up to the part where it builds the username password nonce and message integrity, and the fingerprint.  The realm etc all seem to be good, sent from the server to the client.  Seems all good to me.  But after that there is a 401.  Not sure how to proceed with it.  To my mind that should have succeeded.

P.S i know the spec you get a 401 as first response.  It's not that.

**Comment by ha-ves** [comment]
Is there any error message from coturn/the sipsorcery RTCPeerConnection?

I also use coturn and it works.

**Reply by terraslate** [reply]
Sorry for late reply: holiday. It gets through to CreatePermissionSuccessResponse and then waits for a BindingResponse that never comes presumably because no BindingRequest is sent either from what I can tell. Not sure how to proceed as all appears ok.  Not sure if this makes a difference but the two peers are in the same process.

I set to throw on all exceptions but nothing is thrown.

**Reply by ha-ves** [reply]
I see, the usual culprit is race condition.

It would be better to open issue for better future referencing, if you're able, pls do so.

Also don't forget to include the setup you mentioned.

**Comment by sipsorcery** [comment]
This [comment](https://github.com/sipsorcery-org/sipsorcery/issues/1028#issuecomment-2504873978) might help.

You can use the webrtccmdline program to do a TURN test:

```
sipsorcery\examples\webrtccmdline>dotnet run --ws --stun turn:<your turn server> --relayonly --accepticetypes=relay
```
